Microsoft.Toolkit.Uwp/Helpers/BackgroundTaskHelper.cs

```diff
@@ -0,0 +1,107 @@
+﻿// ******************************************************************
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Windows.ApplicationModel.Background;
+
+namespace Microsoft.Toolkit.Uwp.Helpers
+{
+    /// <summary>
+    /// This class provides static helper methods for background task.
+    /// </summary>
+    public static class BackgroundTaskHelper
+    {
+        /// <summary>
+        /// Check if a background task has been registered with the application or not.
+        /// </summary>
+        /// <param name="backgroundTaskType">The type of the background task.</param>
+        /// <returns>True/False indicating if a background task was registered or not</returns>
+        public static bool IsBackgroundTaskRegistered(Type backgroundTaskType)
+        {
+            return BackgroundTaskRegistration.AllTasks.Any(t => t.Value.Name == backgroundTaskType.Name);
+        }
+
+        /// <summary>
+        /// Register a background task for the application with conditions
+        /// </summary>
+        /// <param name="backgroundTaskType">The type of the background task</param>
+        /// <param name="trigger">Trigger that indicate when the background task should be invoked</param>
+        /// <param name="enforceConditions">Indicate if the background task should quit if condition is no longer valid</param>
+        /// <param name="conditions">Optional conditions for the background task to run with</param>
+        /// <returns>Background Task that was registered with the system</returns>
+        public static async Task<BackgroundTaskRegistration> RegisterAsync(Type backgroundTaskType, IBackgroundTrigger trigger, bool enforceConditions = true, params IBackgroundCondition[] conditions)
+        {
+            // Check if the task is already registered.
+            if (IsBackgroundTaskRegistered(backgroundTaskType) == true)
+            {
+                return GetBackgroundTask(backgroundTaskType) as BackgroundTaskRegistration;
```

Comment:
So we don't want duplicates... I think our options are...

(1) Compare the trigger values and conditions with the existing registration, and if they're different, unregister and re-register

(2) Always unregister and re-register

Option 2 will certainly be easier. But then that gets us into a messy case where, what if a dev has a TimeTrigger task that runs every 24 hours... the user opens the app every day, and the app calls Register each time without checking if it's registered... that would mean we unregister and re-register the task, which means that the task is actually never going to execute, since it keeps on getting registered for 24 hours in the future.

This area is messy lol. I think potentially the cleanest solution is "If there's an existing background task, this method will no-op and simply return." That means if a developer needs to change their trigger, they actually have to unregister and re-register. But then we don't get into this messy complicated state.